File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginalPPMVisit/QuoteContractOriginalPPMVisitSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteContractOriginalPPMVisits

        Public Class QuoteContractOriginalPPMVisitSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function GetAll_For_QuoteContractSiteID(ByVal QuoteContractSiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuoteContractOriginalPPMVisit_GetAll_For_QuoteContractSiteID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuoteContractPPMVisit.ToString
                Return New DataView(dt)
            End Function

            Public Function Insert(ByVal oQuoteContractPPMVisit As QuoteContractOriginalPPMVisit) As QuoteContractOriginalPPMVisit
                _database.ClearParameter()
                AddQuoteContractPPMVisitParametersToCommand(oQuoteContractPPMVisit)

                oQuoteContractPPMVisit.SetQuoteContractPPMVisitID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOriginalPPMVisit_Insert"))
                oQuoteContractPPMVisit.Exists = True
                Return oQuoteContractPPMVisit
            End Function

            Private Sub AddQuoteContractPPMVisitParametersToCommand(ByRef oQuoteContractPPMVisit As QuoteContractOriginalPPMVisit)
                With _database
                    .AddParameter("@QuoteContractSiteID", oQuoteContractPPMVisit.QuoteContractSiteID, True)
                    .AddParameter("@EstimatedVisitDate", oQuoteContractPPMVisit.EstimatedVisitDate, True)
                End With
            End Sub

#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginalPPMVisit/QuoteContractOriginalPPMVisitSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace QuoteContractOriginalPPMVisits
    {
        public class QuoteContractOriginalPPMVisitSQL
        {
            private Sys.Database _database;

            public QuoteContractOriginalPPMVisitSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView GetAll_For_QuoteContractSiteID(int QuoteContractSiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID, true);
                var dt = _database.ExecuteSP_DataTable("QuoteContractOriginalPPMVisit_GetAll_For_QuoteContractSiteID");
                dt.TableName = Sys.Enums.TableNames.tblQuoteContractPPMVisit.ToString();
                return new DataView(dt);
            }

            public QuoteContractOriginalPPMVisit Insert(QuoteContractOriginalPPMVisit oQuoteContractPPMVisit)
            {
                _database.ClearParameter();
                AddQuoteContractPPMVisitParametersToCommand(ref oQuoteContractPPMVisit);
                oQuoteContractPPMVisit.SetQuoteContractPPMVisitID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOriginalPPMVisit_Insert"));
                oQuoteContractPPMVisit.Exists = true;
                return oQuoteContractPPMVisit;
            }

            private void AddQuoteContractPPMVisitParametersToCommand(ref QuoteContractOriginalPPMVisit oQuoteContractPPMVisit)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@QuoteContractSiteID", oQuoteContractPPMVisit.QuoteContractSiteID, true);
                    withBlock.AddParameter("@EstimatedVisitDate", oQuoteContractPPMVisit.EstimatedVisitDate, true);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}